### PR TITLE
DO NOT MERGE - QA for #20555 - Re-style QML Address Bar

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a8bb67f37cae3b9e328fa49227daebe6329b246 available at: https://gist.github.com/434381d7132ed3e51993<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.dmg

**Comment by birarda** [comment]
@ctrlaltdavid looks pretty good - I think we will quickly want to build the error message into the existing bar so we aren't jumping between the two, but I know that you said you would do it this way as in interim step.

The only other thing I notice is that while the fade in looks really good the two elements fade in separately, so you can see part of the address bar peek out behind the circle when it fades in and out. Any way to fix that?

**Comment by birarda** [comment]
One more thing - it looks like it doesn't fade in the first time I pull it up after I've launched - any ideas?

**Comment by ctrlaltdavid** [comment]
@birarda Display the "There is no address information..." in the address bar dialog? Yes, that could be done instead of using a separate dialog. (The mock-ups showed a separate dialog, but showing the message in the address bar dialog would be nicer I think.)

Re fade-in/out showing intersection of circle and rectangle behind it: this is because when the circle is semi-transparent you can see the semi-transparent rectangle behind it. This could be fixed by creating a single SVG background image that incorporates both the circle and the rectangle. I can do this if you'd like.

Re not fading in the first time: I'll have a look.

**Comment by birarda** [comment]
Okay - that's my bad, I forgot the mockup showed the address stuff in a sep dialog. That is great for now.

If you could put those into the same SVG so they fade cleaner that would be great (you can fee in for this, evidently).

**Comment by hifi-gustavo** [comment]
Build console output for commit: b166e24ff5daac0bef3c4d401e7c5d7b6f307f65 available at: https://gist.github.com/434381d7132ed3e51993<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9c52c79fe1df20bcf00ed5ff2c00fa95c3cd8860 available at: https://gist.github.com/434381d7132ed3e51993<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.dmg

File: interface/resources/qml/AddressBarDialog.qml [hunk 1]
```diff
@@ -1,91 +1,152 @@
...
         }
     }
-    
-    Keys.onEscapePressed: {
-        enabled = false;
+
+    // The UI enables an object, rather than manipulating its visibility, so that we can do animations in both directions. 
+	// Because visibility and enabled are booleans, they cannot be animated. So when enabled is changed, we modify a property 
+	// that can be animated, like scale or opacity, and then when the target animation value is reached, we can modify the 
+	// visibility.
```
Comment (birarda) [inline interface/resources/qml/AddressBarDialog.qml]: indentation on comment

**Comment by birarda** [comment]
Looking great - will give the code a review and this is good to go, right?

File: interface/resources/qml/AddressBarDialog.qml [hunk 2]
```diff
@@ -1,91 +1,145 @@
...
+    }
+
+    onVisibleChanged: {
+        if (!visible) {
+            reset()
+
+			// Some dialogs should be destroyed when they become invisible.
+			if (destroyOnInvisible) {
+				destroy()
+			}
```
Comment (birarda) [inline interface/resources/qml/AddressBarDialog.qml]: this is indented farther over than it needs to be

**Comment by ctrlaltdavid** [comment]
Yes, it's good to go (once I've addressed your review items).

**Comment by birarda** [comment]
Looks good to me, a couple of indentation comments

**Comment by ctrlaltdavid** [comment]
A few miscellaneous tabs were messing up the indentation in places.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7d12482a4922d3c84a9371485f87f68112c9bc53 available at: https://gist.github.com/434381d7132ed3e51993<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5012.dmg

**Pull request merged**